controllers/pattern_controller.go

```diff
@@ -180,35 +183,32 @@ func (r *PatternReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		}
 	}
 
-	// -- GitOps Subscription
-	targetSub := newSubscription(*qualifiedInstance)
-	_ = controllerutil.SetOwnerReference(qualifiedInstance, targetSub, r.Scheme)
-
-	_, sub := getSubscription(r.olmClient, targetSub.Name, targetSub.Namespace)
-	if sub == nil {
-		err := createSubscription(r.olmClient, targetSub)
-		return r.actionPerformed(qualifiedInstance, "create gitops subscription", err)
-	} else if ownedBySame(targetSub, sub) {
-		// Check version/channel etc
-		// Dangerous if multiple patterns do not agree, or automatic upgrades are in place...
-		err, changed := updateSubscription(r.olmClient, targetSub, sub)
-		if changed {
-			return r.actionPerformed(qualifiedInstance, "update gitops subscription", err)
-		}
-	} else {
-		logOnce("The gitops subscription is not owned by us, leaving untouched")
+	// Deploy openshift-gitops subscription from config map
+	err = r.deployGitopsSubscription(customConfigFile)
+	if err != nil {
+		fmt.Print("Error: ", err)
+		r.logger.Info("Issue deploying openshift-gitops from configmap")
 	}
```

Comment:
I would be inclined to keep the original logic, and either pass the config map to newSubscription() , or have newSubscription() retrieve it first

Reply:
Agreed.  I will be comparing our current code base.

Reply:
Original logic restored with added subscription function.